example/heatEquation/src/heatEquation.cpp

```diff
@@ -146,8 +146,10 @@ auto example(TAccTag const&) -> int
 
     for(uint32_t step = 0; step < numTimeSteps; step++)
     {
+        auto const& tmpKernelBundle = alpaka::KernelBundle(heatEqKernel, pCurrAcc, pNextAcc, numNodesX, dx, dt);
+
         // Compute next values
-        alpaka::exec<Acc>(queue, workDiv, heatEqKernel, pCurrAcc, pNextAcc, numNodesX, dx, dt);
+        alpaka::exec<Acc>(queue, workDiv, tmpKernelBundle);
```

Comment:
This is one example where I would prefer _not to_ introduce the extra kernel bundle, since it makes the code longer.

Reply:
It uses `getValidWorkDivForKernel` therefore the bundle is required anyway and it is already in the mainline. This changing the kernel call to avoid calling the kernel with different arguments as used in `getValidWorkDivForKernel`

Reply:
It uses `getValidWorkDivForKernel` with one bundle, but it does not use _that_ bundle to launch the kernel, it creates a new one _only_ to launch the kernel with different arguments.

There is no reason to create and use the second bundle.

This is exactly the use case where one should _not_ use a bundle, and call `exec `directly with the arguments.